practices/follow-functional-core-imperative-shell.md

```diff
@@ -3,117 +3,153 @@
-* How might separating business logic from side effects enhance code readability, maintainability, and scalability?
-* In what ways could the Functional Core and Imperative Shell mitigate the impact of changes in infrastructure technology, allowing for smoother transitions and future-proofing our codebase?
-* What benefits might arise from writing unit tests with the assurance that the Functional Core code being tested has no side effects?
-* How could the reduced presence of control statements in the Imperative Shell simplify integration tests?
+* How might separating business logic from I/O enhance code readability, maintainability, and scalability? <ed> Edited "side effects" to "I/O". OK? </ed> 
+* In what ways could the Functional Core, Imperative Shell practice mitigate the impact of changes in infrastructure technology, allowing for smoother transitions and future-proofing our codebase?
+* What benefits might arise from writing unit tests that ensure the functional core code being tested has no side effects?
+* How could the reduced presence of control statements in the imperative shell simplify integration tests?
 
 #### Evaluating Team Readiness
 
-* How prepared are our development teams to embrace a paradigm shift towards functional programming principles?
+* How prepared are our development teams to embrace a paradigm shift toward functional programming principles?
 * Do team members possess the necessary skills and knowledge to implement and maintain code following this practice?
 * What resources, training, or support mechanisms can we provide to facilitate the transition and ensure successful adoption?
 
-#### Transitioning to Functional Core, Imperative Shell Incrementally
+#### Incremental Transition to Functional Core, Imperative Shell 
 
-* How can we identify and prioritize modules or components within our existing codebase that are suitable candidates for transitioning to the Functional Core, Imperative Shell pattern?
-* What strategies can we employ to refactor existing imperative code into pure functions within the functional core while maintaining backward compatibility and minimizing disruptions to ongoing development?
-* Are there opportunities to introduce the Functional Core, Imperative Shell pattern gradually, perhaps starting with new features or modules, before expanding its adoption to legacy code?
+* How can we identify and prioritize modules or components within our existing codebase that are suitable candidates for transitioning to the Functional Core, Imperative Shell pattern? <ed?> I realize we refer to Functional Core, Imperative Shell in many ways in this practice: a practice, an approach, a pattern. Which term is most appropriate to use consistently? Or is it the case that the practice is called Follow Functional Core, Imperative Shell and the pattern is called Functional Core, Imperative Shell? If so, I'll stop nitpicking. ;) </ed> 
```

Comment:
Haha, no worries. It's good to be precise and consistent to avoid confusion. `Functional Core, Imperative Shell` is a pattern created and popularized by Gary Bernhardt. `Follow the Functional Core, Imperative Shell pattern` is the name of the practice we've defined.

Reply:
Got it. Thanks!